src/main/java/io/testproject/sdk/internal/reporting/extensions/junit5/ExceptionsReporter.java

```diff
@@ -0,0 +1,112 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+package io.testproject.sdk.internal.reporting.extensions.junit5;
+
+import io.testproject.sdk.drivers.ReportingDriver;
+import io.testproject.sdk.internal.rest.messages.StepReport;
+import org.junit.jupiter.api.extension.AfterTestExecutionCallback;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.openqa.selenium.WebDriverException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.util.StringUtils;
+
+import java.util.Optional;
+
+/**
+ * JUnit5 extension that takes care of reporting failed assertions.
+ */
+public class ExceptionsReporter implements AfterTestExecutionCallback {
+
+    /**
+     * Logger instance.
+     */
+    private static final Logger LOG = LoggerFactory.getLogger(ExceptionsReporter.class);
+
+    /**
+     * TestProject package name prefix.
+     */
+    public static final String TESTPROJECT_TLD = "io.testproject";
+
+    /**
+     * Report thrown exceptions after test execution.
+     */
+    @Override
+    public void afterTestExecution(final ExtensionContext context) throws Exception {
+
+        Optional<?> obj = context.getTestInstance();
+        if (obj.isEmpty()) {
+            LOG.error("Got an empty object");
+            return;
+        }
+
+        // Check if Test class implements the ExceptionsReporter interface
+        if (!io.testproject.sdk.interfaces.junit5.ExceptionsReporter.class.isAssignableFrom(obj.get().getClass())) {
+            LOG.error("Class [{}] is using extension directly, but instead must implement [{}] interface.",
+                    obj.get().getClass().getName(),
+                    io.testproject.sdk.interfaces.junit5.ExceptionsReporter.class.getName());
+            return;
+        }
+
+        // Get a driver to use for reporting
+        ReportingDriver driver = ((io.testproject.sdk.interfaces.junit5.ExceptionsReporter) obj.get()).getDriver();
+        if (driver == null) {
+            LOG.error("Failed to obtain an instance of [{}] from [{}] to be used for reporting",
+                    ReportingDriver.class.getName(),
+                    ((io.testproject.sdk.interfaces.junit5.ExceptionsReporter) obj.get()).getClass().getName());
+            return;
+        }
+
+        if (context.getExecutionException().isEmpty()) {
+            // Proceed only if an exception was thrown
+            return;
+        }
+
+        if (context.getExecutionException().get() instanceof WebDriverException) {
+            // WebDriverException is already handled by the reporting executor
+            return;
+        }
+
+        Throwable exception = context.getExecutionException().get();
+        if (exception.getClass().getPackage().getName().startsWith(TESTPROJECT_TLD)) {
+            // Do not report SDK internal exceptions
+            return;
+        }

```

Comment:
This code can probably be organised a little differently:

```suggestion
        if (context.getExecutionException().isEmpty()) {
            // Proceed only if an exception was thrown
            return;
        }

        Throwable exception = context.getExecutionException().get();
        
        // Do NOT report WebDriverException as it is already handled by the reporting executor
        // Do NOT report SDK internal exceptions
        if (exception instanceof WebDriverException ||
            exception.getClass().getPackage().getName().startsWith(TESTPROJECT_TLD)) {
            return;
        }
```